specs/interop/derivation.md

```diff
@@ -127,199 +129,212 @@ based upgrade transactions are deterministically generated by the derivation pip
 
-This transaction updates the CrossL2Inbox Proxy ERC-1967
-implementation slot to point to the new CrossL2Inbox deployment.
+This transaction updates the L2ToL2CrossDomainMessenger Proxy ERC-1967
+implementation slot to point to the new L2ToL2CrossDomainMessenger deployment.
 
 A deposit transaction is derived with the following attributes:
 
 - `from`: `0x0000000000000000000000000000000000000000`
-- `to`: `0x4200000000000000000000000000000000000022` (CrossL2Inbox Proxy)
+- `to`: `0x4200000000000000000000000000000000000023` (L2ToL2CrossDomainMessenger Proxy)
 - `mint`: `0`
 - `value`: `0`
 - `gasLimit`: `50,000`
-- `data`: `0x3659cfe6000000000000000000000000691300f512e48b463c2617b34eef1a9f82ee7dbf`
-- `sourceHash`: `0x88c6b48354c367125a59792a93a7b60ad7cd66e516157dbba16558c68a46d3cb`
-  computed with the "Upgrade-deposited" type, with `intent = "Interop: CrossL2Inbox Proxy Update"`
+- `data`: `0x3659cfe60000000000000000000000000d0edd0ebd0e94d218670a8de867eb5c4d37cadd`
+- `sourceHash`: `0xe54b4d06bbcc857f41ae00e89d820339ac5ce0034aac722c817b2873e03a7e68`
+  computed with the "Upgrade-deposited" type, with `intent = "Interop: L2ToL2CrossDomainMessenger Proxy Update"`
 
 Verify data:
 
 ```bash
-cast concat-hex $(cast sig "upgradeTo(address)") $(cast abi-encode "upgradeTo(address)" 0x691300f512e48B463C2617b34Eef1A9f82EE7dBf)
-# 0x3659cfe6000000000000000000000000691300f512e48b463c2617b34eef1a9f82ee7dbf
+cast concat-hex $(cast sig "upgradeTo(address)") $(cast abi-encode "upgradeTo(address)" 0x0D0eDd0ebd0e94d218670a8De867Eb5C4d37cadD)
+# 0x3659cfe60000000000000000000000000d0edd0ebd0e94d218670a8de867eb5c4d37cadd
 ```
 
 Verify `sourceHash`:
 
 ```bash
-cast keccak $(cast concat-hex 0x0000000000000000000000000000000000000000000000000000000000000002 $(cast keccak "Interop: CrossL2Inbox Proxy Update"))
-# 0x88c6b48354c367125a59792a93a7b60ad7cd66e516157dbba16558c68a46d3cb
+cast keccak $(cast concat-hex 0x0000000000000000000000000000000000000000000000000000000000000002 $(cast keccak "Interop: L2ToL2CrossDomainMessenger Proxy Update"))
+# 0xe54b4d06bbcc857f41ae00e89d820339ac5ce0034aac722c817b2873e03a7e68
 ```
 
-### L2ToL2CrossDomainMessenger Deployment
+### L2 Gas Limit Requirements
+
+This upgrade requires `3_270_000` in gas, plus `50_000` gas if the CrossL2Inbox proxy is initialized in the same block.
+In order to upgrade successfully the following invariant MUST be maintained:
+
+```plaintext
+MaxResourceLimit + SystemTxMaxGas + UpgradeGasUsage ≤ L2GasLimit
+```
+
+With MaxResourceLimit set to `20_000_000` and SystemTxMaxGas set to `1_000_000`,
+this mean every upgrading chain MUST have their L2GasLimit set to greater than
+`24_320_000`.
+
+## CrossL2InboxProxy Initialization
+
+On the first block at or after the interop  [activation block](#activation-block), where the current chain is in a
+dependency set with two or more active chains, a deposit transaction based upgrade transaction is deterministically
+generated by the derivation pipeline:
```

Comment:
We don't actually have a way to add chains currently so the implementation is literally just checking this condition at the interop activation block.  We could change the spec to match the more restrictive implementation but I think this version actually says what we want the protocol rules to be, there just isn't a way to add a chain yet so in practice you only need to check the dep set size at the activation block.